diff --git a/lib/vendor/agavi/src/renderer/AgaviSmartyRenderer.class.php b/lib/vendor/agavi/src/renderer/AgaviSmartyRenderer.class.php
index c7d8190..715decc 100644
--- a/lib/vendor/agavi/src/renderer/AgaviSmartyRenderer.class.php
+++ b/lib/vendor/agavi/src/renderer/AgaviSmartyRenderer.class.php
@@ -91,17 +91,7 @@ class AgaviSmartyRenderer extends AgaviRenderer implements AgaviIReusableRendere
 			return $this->smarty;
 		}
 
-		if(!class_exists('Smarty')) {
-			if(defined('SMARTY_DIR') ) {
-				// if SMARTY_DIR constant is defined, we'll use it
-				require(SMARTY_DIR . 'Smarty.class.php');
-			} else {
-				// otherwise we resort to include_path
-				require('Smarty.class.php');
-			}
-		}
-
-		$this->smarty = new Smarty();
+		$this->smarty = $this->createEngine();
 		$this->smarty->clear_all_assign();
 		$this->smarty->clear_config();
 		$this->smarty->config_dir = AgaviConfig::get('core.config_dir');
@@ -128,6 +118,29 @@ class AgaviSmartyRenderer extends AgaviRenderer implements AgaviIReusableRendere
 	}
 
 	/**
+	 * Creates a new smarty instance
+	 *
+	 * @return     Smarty A Smarty instance.
+	 *
+	 * @author     David Zülke <dz@bitxtender.com>
+	 * @since      1.0.2
+	 */
+	protected function createEngine()
+	{
+		if(!class_exists('Smarty')) {
+			if(defined('SMARTY_DIR') ) {
+				// if SMARTY_DIR constant is defined, we'll use it
+				require(SMARTY_DIR . 'Smarty.class.php');
+			} else {
+				// otherwise we resort to include_path
+				require('Smarty.class.php');
+			}
+		}
+
+		return new Smarty();
+	}
+
+	/**
 	 * Render the presentation and return the result.
 	 *
 	 * @param      AgaviTemplateLayer The template layer to render.
