Index: AgaviExecutionFilter.class.php
===================================================================
--- AgaviExecutionFilter.class.php	(revision 1984)
+++ AgaviExecutionFilter.class.php	(working copy)
@@ -246,7 +246,8 @@
 		$request = $this->context->getRequest();
 		
 		$isCacheable = false;
-		if($this->getParameter('enable_caching', true) && is_readable($cachingDotXml = AgaviConfig::get('core.module_dir') . '/' . $moduleName . DIRECTORY_SEPARATOR . 'cache' . DIRECTORY_SEPARATOR . $actionName . '.xml')) {
+		// Allow a global setting to turn off caching (default to on)
+		if($this->getParameter('enable_caching', true) && is_readable($cachingDotXml = AgaviConfig::get('core.module_dir') . '/' . $moduleName . DIRECTORY_SEPARATOR . 'cache' . DIRECTORY_SEPARATOR . $actionName . '.xml') && AgaviConfig::get('core.enable_caching', true)) {
 			// $lm->log('Caching enabled, configuration file found, loading...');
 			// no _once please!
 			include(AgaviConfigCache::checkConfig($cachingDotXml));
