Index: AgaviFileLoggerAppender.class.php
===================================================================
--- AgaviFileLoggerAppender.class.php	(revision 2433)
+++ AgaviFileLoggerAppender.class.php	(working copy)
@@ -70,9 +70,9 @@
 	 */
 	protected function getHandle()
 	{
-		if(is_null($this->handle)) {
-			if(!is_writable(dirname($this->filename)) || (file_exists($this->filename) && !is_writable($this->filename)) || !$this->handle = fopen($this->filename, 'a')) {
-				throw new AgaviLoggingException('Cannot open file "' . $this->filename . '", please check permissions on file or directory.');
+		if($this->handle === null) {
+			if(!($this->handle = @fopen($this->filename, 'a'))) {
+				throw new AgaviLoggingException('Cannot open stream "' . $this->filename . '"; check permissions on file or directory or ensure the given wrapper supports appending');
 			}
 		}
 		return $this->handle;
