Index: libraries/agavi/database/AgaviPropelDatabase.class.php
===================================================================
--- libraries/agavi/database/AgaviPropelDatabase.class.php	(revision 3329)
+++ libraries/agavi/database/AgaviPropelDatabase.class.php	(working copy)
@@ -145,7 +145,7 @@
 		} else {
 			// trigger Propel autoload and go go go
 			if(class_exists('Propel')) {
-				$this->connection = Propel::getConnection();
+				$this->connection = Propel::getConnection($this->getParameter('datasource'));
 				
 				foreach((array)$this->getParameter('init_queries') as $query) {
 					$this->connection->exec($query);
