Index: AgaviEqualsValidator.class.php
===================================================================
--- AgaviEqualsValidator.class.php	(revision 2738)
+++ AgaviEqualsValidator.class.php	(working copy)
@@ -23,6 +23,7 @@
  * 
  * Parameters:
  *   'value'   value which the input should equals to
+ *   'asparam' whether the 'value' should be treated as a parameter name
  *
  * @package    agavi
  * @subpackage validator
@@ -51,6 +52,9 @@
 		// individual arguments that failed
 		if($this->hasParameter('value')) {
 			$value = $this->getParameter('value');
+			if($this->getParameter('asparam', false)) {
+				$value = $this->getData($value);
+			}
 		} else {
 			$value = $this->getData($this->getArgument());
 		}
